src/Std/Time/DateTime/PlainDateTime.lean

```diff
@@ -0,0 +1,587 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.Date
+import Std.Time.Time
+import Std.Time.Internal
+import Std.Time.DateTime.Timestamp
+
+namespace Std
+namespace Time
+open Internal
+
+set_option linter.all true
+
+/--
+Represents a date and time with components for Year, Month, Day, Hour, Minute, Second, and Nanosecond.
+-/
+structure PlainDateTime where
+
+  /--
+  The `Date` component of a `PlainDate`
+  -/
+  date : PlainDate
+
+  /--
+  The `Time` component of a `PlainTime`
+  -/
+  time : PlainTime
+
+  deriving Inhabited, BEq
+
+namespace PlainDateTime
+
+/--
+Converts a `PlainDateTime` to a `Timestamp`
+-/
+def toTimestampAssumingUTC (dt : PlainDateTime) : Timestamp :=
+  let days := dt.date.toDaysSinceUNIXEpoch
+  let nanos := days.toSeconds + dt.time.toSeconds |>.mul 1000000000
+  let nanos := nanos.val + dt.time.nanosecond.val
+  Timestamp.ofNanosecondsSinceUnixEpoch (Nanosecond.Offset.ofInt nanos)
+
```

Comment:
I think that it would be good to have `toTimestamp : PlainDateTime -> ZoneRules -> Timestamp` and `toTimestampWithZone : PlainDateTime -> TimeZone` that basically convert into a `ZonedDateTime` and to a `Timestamp`. The same is true for `PlainDate`.